B-21168/B-21169 INT Location Lookup component

r-mettler opened the pull request:
## [B-21168](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011663)
## [B-21169](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011664)

## Summary

This PR creates a new location lookup component that will be used to search and populate the City State Zip and County fields for the Customer addresses on several pages and the various Office addresses app pages and modals. The fields are being made read only to prevent users from entering bad data into the Address Fields.
An API was added to use to return the results of the search and model was added to use the new v_locations view to query against. 
New addresses added using the city finder will also save/update the us_post_region_cities_id in the addresses table for the address. 

Several pages were modified in the Office App and the Customer Profile creation was updated. The rest of the Customer pages will be updated as part of B-21169 and the Prime Sim will be updated in another backlog.

Is there anything you would like reviewers to give additional scrutiny?

There are many places that were updated to have the new search. Please try and hammer away at any Customer or Office page that uses the <AddressFields> component as there are a LOT of them that were updated. Go through the entire process from making a new Customer to submitting and closing out shipments of all types. Make sure you also include editing of Customer profile and shipment addresses as well as doing this on the Office app side. In short try and break it the best you can. 

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Make sure you can search by just the Postal Code, or by using City, State Postal Code (Note the , after City that is required to help distinguish between City and State). Make sure you can save and continue once you select a result and fill out the rest of the pages. 

#### NOTE: The City State, Zip and County fields should all be read only when the location lookup component is present. 

Test that the us_post_region_cities_id is populated when you add/edit your addresses. To make this task easier I was using the following query each time I modified an address to make sure the us_post_region_cities_id is updated and is not null or all 0's. Note the like, I started all my street address 1 fields with '123' to make the data easier to view. I also recommend you put the name of the page and/or action you are performing when you modify the address to help you keep track of the changes ids. 

#### select a.street_address_1, a.city, a.state, a.postal_code, a.county, a.us_post_region_cities_id from addresses a where a.street_address_1 like '123%%';

1. Access the Customer app and create a new customer profile
2. Test the Customer Address and the Backup Address pages
3. Create a new shipment of various types and verify you can use the search component and submit the shipments
4. Login to the Office App and make sure you can edit customer profile address, edit shipment addresses using the location lookup tool. Try the various shipment types and edit them or add a new shipment and test the location lookup. Do this as TOO and SC to make sure you can edit addresses or add new shipments. 
5. Try searching for Alaska locations when the Alaska FF is off and when its on. You should only see Alaska results when you enable the Alaska FF.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

## Screenshots

r-mettler commented on src/setupTests.js:
```diff
@@ -19,5 +19,5 @@ global.performance = {
 
 jestPreviewConfigure({
-  autoPreview: false,
+  autoPreview: true,
```
This just enables you to run npm jest-preview so you can view the client tests when they fail.

ajlusk replied:
Fair enough, I guess it doesn't really have any effect unless someone is running `npm run jest-preview`. Figured there was a reason it was defaulted to false but I can't think of any.

r-mettler commented:
![image](https://github.com/user-attachments/assets/e1d216bd-2adb-4556-a57a-e69be35a6ffa)
![image](https://github.com/user-attachments/assets/419b404e-f16d-473e-8bc5-4d1932e921d3)

danieljordan-caci commented on pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:
```diff
@@ -2647,3 +2647,31 @@ func ReServiceItems(reServiceItems models.ReServiceItems) ghcmessages.ReServiceI
 }
+
+// VLocation payload
+func VLocation(vLocation *models.VLocation) *ghcmessages.VLocation {
+	if vLocation == nil {
+		return nil
+	}
+	if *vLocation == (models.VLocation{}) {
+		return nil
+	}
+
+	return &ghcmessages.VLocation{
+		City:                 vLocation.CityName,
+		State:                vLocation.StateName,
+		PostalCode:           vLocation.UsprZipID,
+		County:               &vLocation.UsprcCountyNm,
+		UsPostRegionCitiesID: *handlers.FmtUUID(*vLocation.UprcId),
+	}
+}
+
+// VLocations payload
+func VLocations(vLocations models.VLocations) ghcmessages.VLocations {
+	payload := make(ghcmessages.VLocations, len(vLocations))
+	for i, vLocation := range vLocations {
+		copyOfVLocation := vLocation
+		payload[i] = VLocation(&copyOfVLocation)
+	}
+	return payload
+}
```
Please add tests for these

r-mettler replied:
Added

danieljordan-caci commented on pkg/handlers/ghcapi/internal/payloads/payload_to_model.go:
```diff
@@ -952,3 +955,19 @@ func EvaluationReportFromUpdate(evaluationReport *ghcmessages.EvaluationReport)
 	}
 	return &model, nil
 }
+
+func VLocationModel(vLocation *ghcmessages.VLocation) *models.VLocation {
+	if vLocation == nil {
+		return nil
+	}
+
+	usprcID := uuid.FromStringOrNil(vLocation.UsPostRegionCitiesID.String())
+
+	return &models.VLocation{
+		CityName:      vLocation.City,
+		StateName:     vLocation.State,
+		UsprZipID:     vLocation.PostalCode,
+		UsprcCountyNm: *vLocation.County,
+		UprcId:        &usprcID,
+	}
+}
```
Please add a test for this model

r-mettler replied:
Added

danieljordan-caci commented on pkg/handlers/internalapi/internal/payloads/model_to_payload.go:
```diff
@@ -657,3 +657,28 @@ func SignedCertification(signedCertification *models.SignedCertification) *inter
 
 	return model
 }
+
+// VLocation payload
+func VLocation(vLocation *models.VLocation) *internalmessages.VLocation {
+	if vLocation == nil || *vLocation == (models.VLocation{}) {
+		return nil
+	}
+
+	return &internalmessages.VLocation{
+		City:                 vLocation.CityName,
+		State:                vLocation.StateName,
+		PostalCode:           vLocation.UsprZipID,
+		County:               &vLocation.UsprcCountyNm,
+		UsPostRegionCitiesID: *handlers.FmtUUID(*vLocation.UprcId),
+	}
+}
+
+// VLocations payload
+func VLocations(vLocations models.VLocations) internalmessages.VLocations {
+	payload := make(internalmessages.VLocations, len(vLocations))
+	for i, vLocation := range vLocations {
+		copyOfVLocation := vLocation
+		payload[i] = VLocation(&copyOfVLocation)
+	}
+	return payload
+}
```
Please add a test for these models

r-mettler replied:
Added

danieljordan-caci commented on pkg/handlers/internalapi/internal/payloads/payload_to_model.go:
```diff
@@ -21,15 +21,35 @@ func AddressModel(address *internalmessages.Address) *models.Address {
+	}
+}
+
+func VLocationModel(vLocation *internalmessages.VLocation) *models.VLocation {
+	if vLocation == nil {
+		return nil
+	}
+
+	usprcID := uuid.FromStringOrNil(vLocation.UsPostRegionCitiesID.String())
+
+	return &models.VLocation{
+		CityName:      vLocation.City,
+		StateName:     vLocation.State,
+		UsprZipID:     vLocation.PostalCode,
+		UsprcCountyNm: *vLocation.County,
+		UprcId:        &usprcID,
 	}
 }
```
Test plz

r-mettler replied:
Added

danieljordan-caci commented on pkg/services/address/address_lookup.go:
```diff
@@ -0,0 +1,117 @@
+
+	sqlQuery := ""
+
+	if isAlaskaEnabled {
+		sqlQuery = fmt.Sprintf(`
+			select vl.city_name, vl.state, vl.usprc_county_nm, vl.uspr_zip_id, vl.uprc_id
+				from v_locations vl where vl.uspr_zip_id like '%[1]s%%' and
+				vl.city_name like upper('%[2]s%%') and vl.state like upper('%[3]s%%') limit 30`, postalCode, city, state)
+	} else {
+		sqlQuery = fmt.Sprintf(`
+		select vl.city_name, vl.state, vl.usprc_county_nm, vl.uspr_zip_id, vl.uprc_id
+			from v_locations vl where vl.uspr_zip_id like '%[1]s%%' and
+			vl.city_name like upper('%[2]s%%') and vl.state like upper('%[3]s%%') and vl.state NOT in ('AK','HI') limit 30`, postalCode, city, state)
+	}
```
Could you use `RawQuery` instead?
```
    query := appCtx.DB().RawQuery(`
        select vl.city_name, vl.state, vl.usprc_county_nm, vl.uspr_zip_id, vl.uprc_id
        from v_locations vl 
        where vl.uspr_zip_id like ? and
              vl.city_name like upper(?) and 
              vl.state like upper(?)
              ${if !isAlaskaEnabled} and vl.state NOT IN ('AK', 'HI') 
        limit 30`,
        fmt.Sprintf("%s%%", postalCode),
        fmt.Sprintf("%s%%", city),
        fmt.Sprintf("%s%%", state),
    )
```

danieljordan-caci replied:
I say this from a security standpoint (sql injection & all that jazz) - I'd prefer a parameterized approach

deandreJones replied:
security preferts a parameterized approach

r-mettler replied:
Refactored

danieljordan-caci commented on pkg/services/address/address_lookup.go:
```diff
@@ -0,0 +1,117 @@
+	return &vLocation{}
+}
+
+func (o vLocation) GetLocationsByZipCityState(appCtx appcontext.AppContext, search string) (*models.VLocations, error) {
```
This file needs tests

r-mettler replied:
Added

danieljordan-caci commented on src/components/form/AddressFields/AddressFields.jsx:
```diff
@@ -28,46 +30,59 @@ export const AddressFields = ({
+  const assistanceStr = ' for further assistance.';
 
-  let postalCodeField;
+  // const postalCodeField = !locationLookup ? (
+  //   <TextField
+  //     label="ZIP"
+  //     id={`zip_${addressFieldsUUID.current}`}
+  //     name={`${name}.postalCode`}
+  //     data-testid={`${name}.postalCode`}
+  //     maxLength={10}
+  //     labelHint={labelHintProp}
+  //     validate={validators?.postalCode}
+  //   />
+  // ) : (
+  //   <TextField
+  //     label="ZIP"
+  //     id={`zip_${addressFieldsUUID.current}`}
+  //     name={`${name}.postalCode`}
+  //     data-testid={`${name}.postalCode`}
+  //     maxLength={10}
+  //     labelHint={labelHintProp}
+  //     display="locationLookup ? '' : 'readonly'"
+  //     validate={validators?.postalCode}
+  //   />
+  // );
```
remove this comment

r-mettler replied:
Removed

danieljordan-caci reviewed (changes requested):
Couple things and some requested changes:

1. Need to fix the mobile styling here to make the UI elements evenly spaced
![Screenshot 2024-11-21 at 1 39 43 PM](https://github.com/user-attachments/assets/b76ec251-a7c8-4897-95ff-591b147e8ce0)

2. When adding addresses, I'm getting a null `uuid` for my CONUS addresses. Is that expected behavior?
![Screenshot 2024-11-21 at 1 41 44 PM](https://github.com/user-attachments/assets/9d603084-9ca3-4a23-b806-707617467af8)

deandreJones commented on pkg/models/v_locations.go:
```diff
@@ -0,0 +1,44 @@
+country_id IS 'An id of a record in the re_countries table'
+*/
+type VLocation struct {
+	UprcId           *uuid.UUID        `db:"uprc_id" json:"uprc_id" rw:"r"`
```
would like to see some more consistency in this model/ db view
some columns are usprc some have UsPostRegionCity spelled out

r-mettler replied:
I named these based off what the column names were in the DB view. I changed them to use the spelled out version.

r-mettler commented:
> Couple things and some requested changes:
> 
> 1. Need to fix the mobile styling here to make the UI elements evenly spaced

> 2. When adding addresses, I'm getting a null `uuid` for my CONUS addresses. Is that expected behavior?

I had the mobile styling fixed the first time we did this last PI but we changed some things when refactoring. I added the styling back so the spacing is better. 
The id was another thing lost in refactoring, I added it back in as well.

traskowskycaci reviewed (changes requested):
I imagine we want your nice new city finder on the page below, which is when the SC creates a profile on behalf of the customer. To get to this flow, you log in as an SC, go to Customer Search at the top, search for a customer by name (a nonexistent customer), should see a button appear to Add Customer, and then this form that is for creating the customer.

<img width="837" alt="image" src="https://github.com/user-attachments/assets/228ac7f1-ad40-49c2-94ef-5ac9a5ad437c">

<img width="721" alt="image" src="https://github.com/user-attachments/assets/45f998d3-a5fd-4efd-94cd-e9b905ab0784">
<img width="730" alt="image" src="https://github.com/user-attachments/assets/78f27628-807a-40f0-b03b-b3de772ee197">

r-mettler commented:
> I imagine we want your nice new city finder on the page below, which is when the SC creates a profile on behalf of the customer. To get to this flow, you log in as an SC, go to Customer Search at the top, search for a customer by name (a nonexistent customer), should see a button appear to Add Customer, and then this form that is for creating the customer.
> 

Nice find!!! This page wasn't using the original <AddressFields> component. I will go and modify this page.

danieljordan-caci reviewed (changes requested):
Still seeing the null uuids when creating my addresses - should this be null if there is not an associated `us_post_region_cities_id`?

![Screenshot 2024-11-22 at 10 27 13 AM](https://github.com/user-attachments/assets/84d2a464-08d6-4d8b-bba1-df2d0cc6f282)


![Screenshot 2024-11-22 at 10 29 18 AM](https://github.com/user-attachments/assets/1973d61d-becb-48aa-bb2d-07dcb80db85a)

danieljordan-caci commented:
> Still seeing the null uuids when creating my addresses - should this be null if there is not an associated `us_post_region_cities_id`?
> 
> ![Screenshot 2024-11-22 at 10 27 13 AM](https://private-user-images.githubusercontent.com/136510600/389014722-84d2a464-08d6-4d8b-bba1-df2d0cc6f282.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyOTM1MzMsIm5iZiI6MTczMjI5MzIzMywicGF0aCI6Ii8xMzY1MTA2MDAvMzg5MDE0NzIyLTg0ZDJhNDY0LTA4ZDYtNGQ4Yi1iYmExLWRmMmQwY2M2ZjI4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyMlQxNjMzNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NWVkYjRkMzFhYWEwMTA0NmJkZTZmOTc3MTRhMTMwMzliYmJlMzM4ZTIzNzU2MjRjNzE2ZjQwN2ZjNjM2ODExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.d42AyisW7igfslnj-RXs6GoarsgJ8mejwrFWmxMn_ro)
> 
> ![Screenshot 2024-11-22 at 10 29 18 AM](https://private-user-images.githubusercontent.com/136510600/389015512-1973d61d-becb-48aa-bb2d-07dcb80db85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyOTM1MzMsIm5iZiI6MTczMjI5MzIzMywicGF0aCI6Ii8xMzY1MTA2MDAvMzg5MDE1NTEyLTE5NzNkNjFkLWJlY2ItNDhhYS1iYjJkLTA3ZGNiODBkYjg1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyMlQxNjMzNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NTJmYjA4YmM4OGQ1YmM0YTFiNTc2NzE4ZDAxZDUyODU0ZjU2YTExNTUyYzBkZmU5NzhiNDE2MGNmYWQ3M2QxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GXJrz4DQHPsCGeDDLWo2Y_5Dv4fDP8MXvEZt4Ce7FNE)

To add to this - I'm using `74133` and `74137` for my ZIPs. I'm able to find a row when searching `v_locations` but `us_post_region_cities_id` is  isn't populating in the `addresses` table

r-mettler commented:
> I imagine we want your nice new city finder on the page below, which is when the SC creates a profile on behalf of the customer. To get to this flow, you log in as an SC, go to Customer Search at the top, search for a customer by name (a nonexistent customer), should see a button appear to Add Customer, and then this form that is for creating the customer.

I added the city finder to this page and updated the client tests.

danieljordan-caci commented:
> > 74137
> 
> <img alt="image" width="972" src="https://private-user-images.githubusercontent.com/149000222/389602087-5161f62a-1185-4c4d-aeab-ede94361ab89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1NTM3ODgsIm5iZiI6MTczMjU1MzQ4OCwicGF0aCI6Ii8xNDkwMDAyMjIvMzg5NjAyMDg3LTUxNjFmNjJhLTExODUtNGM0ZC1hZWFiLWVkZTk0MzYxYWI4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQxNjUxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OWFiZDg5YTU4NzQxMzYxMDAwYjI1YjNjNTNjMDk2YTUyYTNkYzg4MmZiODIzM2IyYTE4NWQzYWY1M2Y5MTM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.M8UjU-dfzpcmDbvCYRKFG4QBwI0CPShO3yIFuTEZXVo">
> @deandreJones asked me to change the name of the usprcId and usPostRegionCitiesId to be consistent and I missed the renaming in the swagger file. Its saving for me now, try it again. We still need to go through and make changes to actually use the ID from the addresses table as well as all the new tables we added in 21442 so there is still work to be done but at least the ID should be saved as part of the address updates.

Thank ya. I'll take a look. I only emphasize that because I need this data for the pricing proc I'm working on - appreciate your work! This is a chonky mistress.

MInthavongsay commented on pkg/services/address/address_lookup.go:
```diff
@@ -0,0 +1,113 @@
+	FROM v_locations vl where vl.uspr_zip_id like ? AND
+	vl.city_name like upper(?) AND vl.state like upper(?) `
+
+	if !isAlaskaEnabled {
```
I think we want to use the the new FF flag for Hawaii and separate out Hawaii from Alaska.

r-mettler replied:
Made changes to use the Hawaii FF as well. Good call.

r-mettler commented on pkg/handlers/internalapi/internal/payloads/payload_to_model.go:
```diff
@@ -21,15 +21,35 @@ func AddressModel(address *internalmessages.Address) *models.Address {
+		State:              *address.State,
+		PostalCode:         *address.PostalCode,
+		County:             *address.County,
+		UsPostRegionCityId: &usPostRegionCitiesID,
```
There are still a lot of places where we need to update the addresses table to use the **us_post_region_cities_id**, since this is not a requirement for this BL I think we need to just finish this task off when we do all the refactoring for addresses table to use the new tables we added for re_cities, re_states... and lookup using us_post_region_cities_id and tests. Do you concur @danieljordan-caci @deandreJones ?

MInthavongsay replied:
I'm also see an issue where an address is updated using the finder. For example, the saved address is San Francisco, using the the finder to update to Los Angeles, I noticed us_post_region_cities_id field is not updating to the updated to the new postal code. The database is still using the old us_post_region_cities_id value.

danieljordan-caci reviewed (changes requested):
`us_post_region_cities_id` not filling in - getting `NULL` and a nil `uuid` value when checking the `Use my current address` box when creating shipments. Considering we need these values for downstreams and this implementation directly impacts that logic and populating those values... I think this BL is a good place to include that logic.

Steps to recreate:
- customer add shipment
- check `Use my current address`
- say you don't know your destination addy
- fill out rest of the form
- hit submit
- check db
- `NULL` for the destination address
- nil `uuid` for the pickup

![Screenshot 2024-11-25 at 11 47 32 AM](https://github.com/user-attachments/assets/907c47f6-b458-4534-a90e-2ccaecf21292)

r-mettler commented:
> `us_post_region_cities_id` not filling in - getting `NULL` and a nil `uuid` value when checking the `Use my current address` box when creating shipments. Considering we need these values for downstreams and this implementation directly impacts that logic and populating those values... I think this BL is a good place to include that logic.
> 
> Steps to recreate:
> 
> * customer add shipment
> * check `Use my current address`
> * say you don't know your destination addy
> * fill out rest of the form
> * hit submit
> * check db
> * `NULL` for the destination address
> * nil `uuid` for the pickup
> 

When you check use current address its not using the city finder logic so the us_post_region_cities_id is not saved because the id isn't being set. I only have it set as part of the handler when you select a result for the city finder search. We had talked about doing separate BL to add in all the code and test changes for adding in the use of the us_post_region_cities_id as separate BL during an IP sprint. I hate to add even more to this BL since its so large. Should we just create another BL to address the rest of the populating of the us_post_region_cities_id as it sounds like something may have changed since our discussion to make this connection during IP sprint BL? 
To add it in when you click the use current address checkbox you would have to go in and change to have it loaded into the Customer object for the pages when it loads the page, which isn't related to city finder and to me is out of scope for this BL. We should do this in a different BL.

danieljordan-caci commented:
> > `us_post_region_cities_id` not filling in - getting `NULL` and a nil `uuid` value when checking the `Use my current address` box when creating shipments. Considering we need these values for downstreams and this implementation directly impacts that logic and populating those values... I think this BL is a good place to include that logic.
> > Steps to recreate:
> > 
> > * customer add shipment
> > * check `Use my current address`
> > * say you don't know your destination addy
> > * fill out rest of the form
> > * hit submit
> > * check db
> > * `NULL` for the destination address
> > * nil `uuid` for the pickup
> 
> When you check use current address its not using the city finder logic so the us_post_region_cities_id is not saved because the id isn't being set. I only have it set as part of the handler when you select a result for the city finder search. We had talked about doing separate BL to add in all the code and test changes for adding in the use of the us_post_region_cities_id as separate BL during an IP sprint. I hate to add even more to this BL since its so large. Should we just create another BL to address the rest of the populating of the us_post_region_cities_id as it sounds like something may have changed since our discussion to make this connection during IP sprint BL? To add it in when you click the use current address checkbox you would have to go in and change to have it loaded into the Customer object for the pages when it loads the page, which isn't related to city finder and to me is out of scope for this BL. We should do this in a different BL.

I'll leave it up to @deandreJones - cool either way as long as we are tracking the work and know that we need to get it done ASAP as possible.

danieljordan-caci commented:
Is there a BL addressing the Prime Sim for this ZIP finder? Or are we bypassing that?

r-mettler commented:
> Is there a BL addressing the Prime Sim for this ZIP finder? Or are we bypassing that?

B-21170, Alex has it up for draft but this has to go in first

danieljordan-caci commented:
> > Is there a BL addressing the Prime Sim for this ZIP finder? Or are we bypassing that?
> 
> B-21170, Alex has it up for draft but this has to go in first

Overall it looks pretty good. My only concern is the inconsistent saving of `us_post_region_cities_id` throughout. Works in some forms, doesn't work in others. If there's a BL that gets created to handle that functionality and fill the gap that this leaves - I'm cool with this. But I don't want to approve until we make a decision on that.

danieljordan-caci commented on src/pages/Office/CustomerOnboarding/CreateCustomerForm.module.scss:
```diff
@@ -11,3 +16,42 @@
+  }
+}
+
+// .form {
+//   h2 {
+//     @include u-margin-bottom(0);
+//   }
+
+//   .formSection {
+//     p {
+//       @include u-margin-bottom(0);
+//     }
+
+//     > :global(.usa-form-group:first-of-type .usa-label),
+//     > :global(.usa-fieldset > .usa-form-group:first-of-type .usa-label) {
+//       @include u-margin-top(3);
+//     }
+
+//     > :global(.usa-fieldset > .usa-form-group.usa-form-group--error:first-of-type .usa-label) {
+//       @include u-margin-top(0);
+//     }
+//   }
+// }
```
left in some commented out code

r-mettler replied:
oopsie..I'll go remove that

r-mettler replied:
removed

danieljordan-caci commented on swagger-def/ghc.yaml:
```diff
@@ -4021,6 +4022,31 @@ paths:
           $ref: '#/responses/NotFound'
         '500':
           $ref: '#/responses/ServerError'
+  /addresses/zip_city_lookup/{search}:
```
Can you change this route to use `-` instead of `_`? Majority of the routes use `-` instead of `_`. It also really bugs me that we have inconsistencies throughout this file regarding that, but going with the majority seems to be my preference.

r-mettler replied:
changed to use -

danieljordan-caci commented on swagger-def/internal.yaml:
```diff
@@ -4425,7 +4429,32 @@ paths:
         '500':
           description: server error
-  '/mto_shipments':
+  /addresses/zip_city_lookup/{search}:
```
`-` instead of `_`

r-mettler replied:
changed to use -

danieljordan-caci commented on swagger-def/internal.yaml:
```diff
@@ -4425,7 +4429,32 @@ paths:
+          description: the requested list of city, state, county, and postal code matches
+          schema:
+            $ref: "#/definitions/VLocations"
+        "400":
+          description: invalid request
+        "403":
+          description: not authorized
+        "404":
+          description: not found
+        "500":
+          description: server error
```
Can you return response objects instead of just a description?
Example:
```
        '403':
          $ref: '#/responses/PermissionDenied'
```

r-mettler replied:
changed

danieljordan-caci commented on swagger-def/internal.yaml:
```diff
@@ -4425,7 +4429,32 @@ paths:
-  '/mto_shipments':
+  /addresses/zip_city_lookup/{search}:
+    get:
+      summary: Returns City, State, Postal Code, and County associated with the specified full/partial Postal Code or City State string
+      description: Find by API using full/partial Postal Code or City name that returns an us_post_region_cities json object containing city, state, county and postal code.
```
Inconsistencies with your capitalizations. I'd prefer:
```
Returns city, state, postal code, and county associated with the specified full/partial postal code, city and/or state string
```

and

```
Find by API using full/partial postal code or city name that returns a `us_post_region_cities` json object containing city, state, county and postal code.
```

r-mettler replied:
changed

danieljordan-caci commented on swagger-def/definitions/VLocation.yaml:
```diff
@@ -0,0 +1,129 @@
+  county:
+    type: string
+    title: County
+    x-nullable: true
```
`county` should not be `x-nullable: true`

In the `addresses` model it is not null:
```
County             string            `json:"county" db:"county"`
```

r-mettler replied:
actually this is correct addresses model needs to have county as a *string because it can be null and will be when we add in OCONUS addresses. I'm changing this now.

danieljordan-caci replied:
Yikes. Have fun with those changes lol

r-mettler replied:
using *string now since County can be null and updated code and tests as well to use the pointer

danieljordan-caci commented on swagger-def/definitions/UsPostRegionCity.yaml:
```diff
@@ -0,0 +1,126 @@
+description: A postal code and city lookup
```
Should this also contain `Country`? Looking at the `us_post_regions_cities` table there is a `ctry_genc_dgph_cd` aka "country" column.

Also is there a reason we are using `postalCode` instead of `ZIP`? Just trying to keep the db & yaml 1:1

r-mettler replied:
This file is actually no longer needed since I am using in the VLocation.yaml now. Removing this file.

r-mettler commented on swagger-def/ghc.yaml:
```diff
@@ -4021,6 +4022,31 @@ paths:
+        "404":
+          description: not found
+        "500":
+          description: server error
```
changed

r-mettler commented on swagger-def/ghc.yaml:
```diff
@@ -4021,6 +4022,31 @@ paths:
+  /addresses/zip_city_lookup/{search}:
+    get:
+      summary: Returns City, State, Postal Code, and County associated with the specified full/partial Postal Code or City and State string
+      description: Find by API using full/partial Postal Code or City name that returns an us_post_region_cities json object containing city, state, county and postal code.
```
fixed

r-mettler commented:
> > > Is there a BL addressing the Prime Sim for this ZIP finder? Or are we bypassing that?
> > 
> > 
> > B-21170, Alex has it up for draft but this has to go in first
> 
> Overall it looks pretty good. My only concern is the inconsistent saving of `us_post_region_cities_id` throughout. Works in some forms, doesn't work in others. If there's a BL that gets created to handle that functionality and fill the gap that this leaves - I'm cool with this. But I don't want to approve until we make a decision on that.

Asked Wendy to create a feature that will hold this BL as well as the many others we need to do to update the code and tests to use the new DB design we introduced with B-21442.

MInthavongsay commented on src/components/LocationSearchBox/LocationSearchBox.jsx:
```diff
@@ -179,7 +209,7 @@ export const LocationSearchBoxComponent = ({
   };
 
-  const noOptionsMessage = () => (inputValue.length ? 'No Options' : '');
+  const noOptionsMessage = () => (inputValue.length ? 'No Options' : 'No Options');
```
This logic means either way, it's going to be 'No Options'.  Was this a copy/paste error?

r-mettler replied:
I blame git merge because I didn't make that change. I reverted it

MInthavongsay commented on pkg/handlers/ghcapi/internal/payloads/payload_to_model.go:
```diff
@@ -92,10 +92,13 @@ func AddressModel(address *ghcmessages.Address) *models.Address {
 		return nil
 	}
 
+	usPostRegionCitiesID := uuid.FromStringOrNil(address.UsPostRegionCitiesID.String())
```
Even though you are using fromStringOrNil, you may run into a nil pointer ex on address.UsPostRegionCitiesID.String() if UsPostRegionCitiesID is nil when you do a tostring on it.

r-mettler commented:
> > > Is there a BL addressing the Prime Sim for this ZIP finder? Or are we bypassing that?
> > 
> > 
> > B-21170, Alex has it up for draft but this has to go in first
> 
> Overall it looks pretty good. My only concern is the inconsistent saving of `us_post_region_cities_id` throughout. Works in some forms, doesn't work in others. If there's a BL that gets created to handle that functionality and fill the gap that this leaves - I'm cool with this. But I don't want to approve until we make a decision on that.

I believe I have all of the pages for Customer and Office app saving and updating the us_post_region_cities_id in the addresses table now. You can also use the "Use current address" check boxes and it will save the us_post_region_cities_id. Please test and see the amended test instructions for a helpful query to use for testing. This should work for all addresses the city finder is used to update. One to check that I have not yet is the Storage Facility addresses as part of the NTS-Release shipments. I tested HHG and PPM adding and editing as well as Customer profile first creation and editing.

danieljordan-caci commented on pkg/handlers/authentication/devlocal.go:
```diff
@@ -488,13 +488,16 @@ func createUser(h devlocalAuthHandler, w http.ResponseWriter, r *http.Request) (
 		return nil, userType
 	}
 
+	usPostRegionCityID := uuid.Must(uuid.NewV4())
```
Shouldn't we populate the actual `uuid` of something in the `us_post_region_cities` table here?

danieljordan-caci reviewed (changes requested):
Attaching a recording:
https://github.com/user-attachments/assets/9444e765-6a56-4319-891e-f100ce0ba41f

The validation for the shipment create form seems to be causing an issue. I'm not sure if this is because of your changes, but when I click the "yes" option when filling out a delivery address, fill out the street address 1, but then go to click "no" (a valid use case imo), the validation is failing and won't let me submit the form.

Going to continue to test and seeing lots of good stuff, but this is the first weird thing I encountered. If this is unrelated to your work, I can open an issue.

danieljordan-caci reviewed (changes requested):
Adding another recording:
https://github.com/user-attachments/assets/4044d14e-6d81-4458-839e-9169498d0e59

I have the speed throttled to Slow 4G to show what I'm talking about. The validations seem to run despite the values being present.

ajlusk commented:
> Attaching a recording: https://github.com/user-attachments/assets/9444e765-6a56-4319-891e-f100ce0ba41f
> 
> The validation for the shipment create form seems to be causing an issue. I'm not sure if this is because of your changes, but when I click the "yes" option when filling out a delivery address, fill out the street address 1, but then go to click "no" (a valid use case imo), the validation is failing and won't let me submit the form.
> 
> Going to continue to test and seeing lots of good stuff, but this is the first weird thing I encountered. If this is unrelated to your work, I can open an issue.

Just tested it and this is occurring on the latest main so is unconnected to these changes.

traskowskycaci reviewed (approved):
Scared to be the first one to accept this but here goes

Couple comments from a few folks to address but I haven't been able to break anything!

r-mettler commented:
> > Adding another recording: https://github.com/user-attachments/assets/4044d14e-6d81-4458-839e-9169498d0e59
> > I have the speed throttled to Slow 4G to show what I'm talking about. The validations seem to run despite the values being present.
> 
> @r-mettler Removing the `setFieldTouched()` calls from AddressFields.jsx seems to fix this for me. I haven't found removing them to cause any issues but haven't check _every_ form.

> > Adding another recording: https://github.com/user-attachments/assets/4044d14e-6d81-4458-839e-9169498d0e59
> > I have the speed throttled to Slow 4G to show what I'm talking about. The validations seem to run despite the values being present.
> 
> @r-mettler ~Removing the `setFieldTouched()` calls from AddressFields.jsx seems to fix this for me. I haven't found removing them to cause any issues but haven't check _every_ form.~ Nevermind

@danieljordan-caci The issue turned out to be how Formik was doing its validation. We changed it to do the validation of the fields after all the values have been changed and touched. Try it again now, @ajlusk and I don't see the `Required` label on slow 4G anymore.

danieljordan-caci reviewed (changes requested):
Maaaaan I was so close to approving, but there's a weird font weight applied to the delivery address part when creating a shipment as an office user (specifically NTS-r in my example)
Regular
<img width="568" alt="Screenshot 2024-12-03 at 4 23 14 PM" src="https://github.com/user-attachments/assets/c635cfd0-c733-42cb-9829-152dcbce57a0">

This bold gurl
<img width="605" alt="Screenshot 2024-12-03 at 4 23 46 PM" src="https://github.com/user-attachments/assets/3e98a633-bcfd-4a5e-9be8-c330b50babb3">

danieljordan-caci reviewed (changes requested):
Also some inconsistent styling/spacing on the SC create customer form between county & state

<img width="726" alt="Screenshot 2024-12-03 at 4 30 31 PM" src="https://github.com/user-attachments/assets/5067dc87-7a1f-4b36-b67a-310653239cdc">

<img width="754" alt="Screenshot 2024-12-03 at 4 31 10 PM" src="https://github.com/user-attachments/assets/3f981ae2-8d1e-4d54-b5ed-3960d2408736">

robot-mymove commented:
# Test coverage on server failed.
  **Last run: Wed 04 Dec 2024 04:43:38 PM UTC**
  Refer to [this report](https://output.circle-artifacts.com/output/job/a89ef468-7de1-4701-9ace-f359eb820a5b/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app.
  Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

r-mettler commented:
> Maaaaan I was so close to approving, but there's a weird font weight applied to the delivery address part when creating a shipment as an office user (specifically NTS-r in my example) Regular 
> This bold gurl 

@danieljordan-caci, @ajlusk helped me fix this it was a styling applied by mockLegends style. Fixed now.

robot-mymove commented:
# Test coverage on server failed.
  **Last run: Wed 04 Dec 2024 05:14:27 PM UTC**
  Refer to [this report](https://output.circle-artifacts.com/output/job/41c66215-1a4b-48fa-8079-dcd483bf7161/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app.
  Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

robot-mymove commented:
# Test coverage on server failed.
  **Last run: Wed 04 Dec 2024 08:45:31 PM UTC**
  Refer to [this report](https://output.circle-artifacts.com/output/job/183c21d0-1280-44cb-ba6d-8650889f58df/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app.
  Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

danieljordan-caci commented:
> > Also some inconsistent styling/spacing on the SC create customer form between county & state
> 
> @danieljordan-caci I had noticed this and Cory and I were trying to fix this but I believe this goes back to the same issue that you were looking into with the styling issues of the `.usa-form` and how its applied for mobile. Did you figure out a way to fix this?
> 
> https://caci-bits.slack.com/archives/C05DX3AG6EL/p1731511406976919?thread_ts=1731510553.780619&cid=C05DX3AG6EL

I did! You have to find the styling that's being applied and override it in `_overrides.scss`

danieljordan-caci commented:
> > Also some inconsistent styling/spacing on the SC create customer form between county & state
> 
> @danieljordan-caci I had noticed this and Cory and I were trying to fix this but I believe this goes back to the same issue that you were looking into with the styling issues of the `.usa-form` and how its applied for mobile. Did you figure out a way to fix this?
> 
> https://caci-bits.slack.com/archives/C05DX3AG6EL/p1731511406976919?thread_ts=1731510553.780619&cid=C05DX3AG6EL

I don't believe they're the same issue. This specific spot is because of this line in `AddressFields.jsx`
```
<div className="mobile-lg:grid-col-6" style={{ marginBottom: '1.5em' }}>
```

Also can we avoid using inline stylings like that and instead add it to `AddressFields.module.scss`?

r-mettler commented:
> > > Also some inconsistent styling/spacing on the SC create customer form between county & state
> > 
> > 
> > @danieljordan-caci I had noticed this and Cory and I were trying to fix this but I believe this goes back to the same issue that you were looking into with the styling issues of the `.usa-form` and how its applied for mobile. Did you figure out a way to fix this?
> > https://caci-bits.slack.com/archives/C05DX3AG6EL/p1731511406976919?thread_ts=1731510553.780619&cid=C05DX3AG6EL
> 
> I don't believe they're the same issue. This specific spot is because of this line in `AddressFields.jsx`
> 
> ```
> <div className="mobile-lg:grid-col-6" style={{ marginBottom: '1.5em' }}>
> ```
> 
> Also can we avoid using inline stylings like that and instead add it to `AddressFields.module.scss`?

The real problem with that page is that for some reason its loading the city state sections in the format it should for mobile and if you change to mobile styling in the inspector its loading how it should for desktop. I am not sure why only this page is doing that.

danieljordan-caci commented:
> > > Also some inconsistent styling/spacing on the SC create customer form between county & state
> > 
> > 
> > @danieljordan-caci I had noticed this and Cory and I were trying to fix this but I believe this goes back to the same issue that you were looking into with the styling issues of the `.usa-form` and how its applied for mobile. Did you figure out a way to fix this?
> > https://caci-bits.slack.com/archives/C05DX3AG6EL/p1731511406976919?thread_ts=1731510553.780619&cid=C05DX3AG6EL
> 
> I don't believe they're the same issue. This specific spot is because of this line in `AddressFields.jsx`
> 
> ```
> <div className="mobile-lg:grid-col-6" style={{ marginBottom: '1.5em' }}>
> ```
> 
> Also can we avoid using inline stylings like that and instead add it to `AddressFields.module.scss`?

Removing that does cause some issues with mobile styling per this screenshot:
![Screenshot 2024-12-06 at 10 52 53 AM](https://github.com/user-attachments/assets/f15407aa-c747-4419-845d-8e9d7e57965a)

You'll need to remove that and when adding your style to `AddressFields.module.scss` just add this to it, which will only apply when they're detecting mobile is being used:
```
  @media (min-width: $tablet) {
    .styleName {
      margin-bottom: 1.5rem;
    }
```

r-mettler commented:
> > > > Also some inconsistent styling/spacing on the SC create customer form between county & state
> > > 
> > > 
> > > @danieljordan-caci I had noticed this and Cory and I were trying to fix this but I believe this goes back to the same issue that you were looking into with the styling issues of the `.usa-form` and how its applied for mobile. Did you figure out a way to fix this?
> > > https://caci-bits.slack.com/archives/C05DX3AG6EL/p1731511406976919?thread_ts=1731510553.780619&cid=C05DX3AG6EL
> > 
> > 
> > I don't believe they're the same issue. This specific spot is because of this line in `AddressFields.jsx`
> > ```
> > <div className="mobile-lg:grid-col-6" style={{ marginBottom: '1.5em' }}>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also can we avoid using inline stylings like that and instead add it to `AddressFields.module.scss`?
> 
> Removing that does cause some issues with mobile styling per this screenshot: ![Screenshot 2024-12-06 at 10 52 53 AM](https://private-user-images.githubusercontent.com/136510600/393336999-f15407aa-c747-4419-845d-8e9d7e57965a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MDYwNTgsIm5iZiI6MTczMzUwNTc1OCwicGF0aCI6Ii8xMzY1MTA2MDAvMzkzMzM2OTk5LWYxNTQwN2FhLWM3NDctNDQxOS04NDVkLThlOWQ3ZTU3OTY1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNlQxNzIyMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDAzNGJhZjRiYjE2YTQxM2U4NmEyYmQ4NGY3NjUwMGZhODJiMjc3Y2FhMmEzMjAyNzY0OTY3MjcxZTVhNDQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Le-_fZNR8WvFx8FtEdrrdto8lrtQMziB3o4O8xjU37s)
> 
> You'll need to remove that and when adding your style to `AddressFields.module.scss` just add this to it, which will only apply when they're detecting mobile is being used:
> 
> ```
>   @media (min-width: $tablet) {
>     .styleName {
>       margin-bottom: 1.5rem;
>     }
> ```

Made some modifications to the styling to remove the mobile-lg since that seemed to cause problems. With this new styling the create customer page as the SC doesn't display mobile layout (1 column) by default. I also changed the order of the columns to more closely match the format of the results, city state zip then county name.

r-mettler commented:
> What a ride. SEND IT!

I know right!! So many changes. Appreciate the thoroughness, this will be a great addition to the application.

deandreJones commented:
Happo is unhappy

ajlusk commented:
Lots of false positives in Happo due to pre-existing issues on INT and that this is being compared to main but these ones look like they could be a result of this branch:
[`Date and Location Form` and the three components below it](https://happo.io/a/488/p/603/compare/8d3fcf48049ca050ec93e6d1c571f6252cd2cbe7/80fee29c96d23bfb0f5f922adc892d62ddbae10e#Customer%5C%20Components%5C%20%5C%2F%5C%20PPM%5C%20Booking%5C%20%5C%2F%5C%20Date%5C%20and%5C%20Location%5C%20Form-Blank%5C%20Dates%5C%20And%5C%20Location)
[`Storage Facility Address`](https://happo.io/a/488/p/603/compare/8d3fcf48049ca050ec93e6d1c571f6252cd2cbe7/80fee29c96d23bfb0f5f922adc892d62ddbae10e#Customer%5C%20Components%5C%20%5C%2F%5C%20PPM%5C%20Booking%5C%20%5C%2F%5C%20Date%5C%20and%5C%20Location%5C%20Form-Blank%5C%20Dates%5C%20And%5C%20Location)

r-mettler commented:
> Lots of false positives in Happo due to pre-existing issues on INT and that this is being compared to main but these ones look like they could be a result of this branch: [Date and Location Form and the three components below it](https://happo.io/a/488/p/603/compare/8d3fcf48049ca050ec93e6d1c571f6252cd2cbe7/80fee29c96d23bfb0f5f922adc892d62ddbae10e#Customer%5C%20Components%5C%20%5C%2F%5C%20PPM%5C%20Booking%5C%20%5C%2F%5C%20Date%5C%20and%5C%20Location%5C%20Form-Blank%5C%20Dates%5C%20And%5C%20Location) [Storage Facility Address](https://happo.io/a/488/p/603/compare/8d3fcf48049ca050ec93e6d1c571f6252cd2cbe7/80fee29c96d23bfb0f5f922adc892d62ddbae10e#Customer%5C%20Components%5C%20%5C%2F%5C%20PPM%5C%20Booking%5C%20%5C%2F%5C%20Date%5C%20and%5C%20Location%5C%20Form-Blank%5C%20Dates%5C%20And%5C%20Location) [Edit Facility Info Modal and one below it](https://happo.io/a/488/p/603/compare/8d3fcf48049ca050ec93e6d1c571f6252cd2cbe7/80fee29c96d23bfb0f5f922adc892d62ddbae10e#Office%5C%20Components%5C%2FEditFacilityInfoModal)
> 
> I believe the rest are either approvable or unrelated.

I fixed the Happo files that were complaining about not having the <Provider> for the redux store. Should be good now.

r-mettler commented:
> Happo is unhappy

Happo isn't the only one unhappy 😄

deandreJones reviewed (approved):
heerrree weee gooo
